py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,750 @@
+    wind_unit = 'mph'
+
+    def _get_icons(self):
+        if self.icons is None:
+            self.icons = {}
+
+        # Defaults for weather ranges
+        defaults = {
+            'i200_i299': self.icon_thunderstorm,
+            'i300_i399': self.icon_rain,
+            'i500_i599': self.icon_rain,
+            'i600_i699': self.icon_snow,
+            'i700_i799': self.icon_atmosphere,
+            'i800': self.icon_sun,
+            'i801_i809': self.icon_cloud,
+            'i900_i909': self.icon_extreme,
+            'i950_i959': self.icon_wind,
+            'i960_i999': self.icon_extreme,
+        }
+
+        # Default mappings for other icons
+        others = {
+            'clouds': 802,
+            'rain': 501,
+            'snow': 601,
+            'wind': 954,
+            'humidity': self.icon_humidity,
+            'pressure': self.icon_pressure,
+            'temperature': self.icon_temperature,
+            'sunrise': 800,
+            'sunset': 801,
+        }
+
+        # Handling ranges from OpenWeatherMap
+        data = {}
+        for source in (defaults, self.icons):
+            for key in source:
+                if (key[0] != 'i' and key not in others):
+                    raise Exception('Icon identifier is invalid! (%s)' % key)
+
+                if key[0] == 'i':
+                    if '_' in key:
+                        if key.count('_i') != 1:
+                            raise Exception('Icon range is not properly'
+                                            'formatted! (%s)' % key)
+
+                        # Populate each code
+                        (start, end) = tuple(map(int, key[1:].split('_i')))
+                        for code in range(start, end + 1):
+                            data[code] = source[key]
+
+                    else:
+                        data[int(key[1:])] = source[key]
+                else:
+                    data[key] = source[key]
+
+        # Weather icons for formatting sections
+        for key in others:
+            if key not in data:
+                if isinstance(others[key], int):
+                    data[key] = data[others[key]]
+                else:
+                    data[key] = others[key]
+
+        return data
+
+    def post_config_hook(self):
+        # Verify the API key
+        if self.api_key is None:
+            raise OWMException('API Key for OpenWeatherMap cannot be empty!'
+                               ' Go to http://openweathermap.org/appid to'
+                               ' get an API Key.')
+
+        # Generate our icon array
+        self.icons = self._get_icons()
+
+        # Verify the units configuration
+        if self.rain_unit not in RAIN_UNITS:
+            raise Exception('rain_unit is not recognized')
+        if self.snow_unit not in SNOW_UNITS:
+            raise Exception('snow_unit is not recognized')
+        if self.temp_unit not in TEMP_UNITS:
+            raise Exception('temp_unit is not recognized')
```

Comment:
Fixed! I let any capitalization work for units, so this should not be a problem :)